pkg/controller/controller.go

```diff
@@ -104,8 +149,25 @@ func GetAllTopControllers(ctx context.Context, dynamicClient dynamic.Interface,
 		existingWorkload, ok := workloadMap[key]
 		if !ok {
 			existingWorkload.TopController = controller
+			podSpec, err := GetPodSpec(controller.UnstructuredContent())
+			if err != nil {
+				return nil, err
```

Comment:
Is return appropriate here? Don't we want to keep going through the pod loop? (same below)